cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraCluster.java

```diff
@@ -306,23 +318,31 @@ public TasksForOffer getTasksForOffer(@NotNull final Protos.Offer offer) {
 
             CassandraNode.Builder node;
             if (!nodeOption.isPresent()) {
+                if (clusterState.get().getNodesToAcquire() <= 0) {
+                    // number of C* cluster nodes already present
+                    return null;
+                }
+
                 NodeCounts nodeCounts = clusterState.nodeCounts();
                 if (nodeCounts.getNodeCount() >= defaultConfigRole.getNumberOfNodes()) {
                     // number of C* cluster nodes already present
                     return null;
                 }
 
+                String replacementForIp = clusterState.nextReplacementIp();
+
                 boolean buildSeedNode = nodeCounts.getSeedCount() < defaultConfigRole.getNumberOfSeeds();
-                CassandraNode newNode = buildCassandraNode(offer, buildSeedNode);
-                clusterState.nodes(append(
-                        clusterState.nodes(),
-                        newNode
-                ));
+                CassandraNode newNode = buildCassandraNode(offer, buildSeedNode, replacementForIp);
+                clusterState.nodeAcquired(newNode);
                 node = CassandraNode.newBuilder(newNode);
             } else
                 node = CassandraNode.newBuilder(nodeOption.get());
 
             if (!node.hasCassandraNodeExecutor()) {
+                if (node.getTargetRunState() == CassandraNode.TargetRunState.TERMINATE) {
```

Comment:
Currently, yes. We should "lock out" the terminated and replaced node for some days until the last node in the cluster has forgotten the replaced node. (It may take some time until the last node recognizes that - such a "last" node could be one that has been dead for some time, too.)

Sure, we re-check such stuff as soon as we have permanently allocated resources from Mesos (get the same data volumes and ports from Mesos after a restart of the executor)